Run all given experiments

Zelldon opened the pull request:
Turned out that previous we only executed the first experiment. :-1: 
This PR fixes this problem and adds more output to the job completion, which shows which experiments have been executed successfully. Added new unit tests for that.

Furthermore as suggested if experiments are skipped I also added a new output and test result. Might need to adjust also the java code, but not 100% sure whether this is necessary, since we only check the result via FEEL and check whether is has failed.

Zelldon commented:
Nice first time the build failed and looks quite good, the output :D It was because I rebased with master, missed to update a test case.

Zelldon commented:
New output:

```sh
+ zbctl --insecure create worker chaos-experiments --concurrency 1 --handler handleJob.sh
2020/11/12 16:18:03 Activated job 2251799813685585 with variables {"clusterId":"810162a9-1a38-41d0-a358-084037e1cd51","testResult":{},"clusterPlan":"Production - M","authenticationDetails":{"authorizationURL":"https://login.cloud.ultrawombat.com/oauth/token","contactPoint":"810162a9-1a38-41d0-a358-084037e1cd51.zeebe.ultrawombat.com:443","clientId":"-NFYPbF8hFpfuLusX0JxxhJcEfd-cNbN","clientSecret":"oFTCr4d4tv61X8eYDGnbiDRl_OrHq.wMtw7ZMXuI4obBM-BLbg4QmjCCWnkwGKTG"}}
2020/11/12 16:29:06 Handler completed job 2251799813685585 with variables {
  "testResult": "PASSED",
  "testReport": {
    "testResult": "PASSED",
    "failureMessages": [],
    "failureCount": 0,
    "metaData": {
      "results": [
        "production-m/follower-restart/experiment.json run successfully",
        "production-m/follower-terminate/experiment.json run successfully",
        "production-m/high-load/experiment.json run successfully",
        "production-m/leader-restart/experiment.json run successfully",
        "production-m/leader-terminate/experiment.json run successfully",
        "production-m/multiple-leader-restart/experiment.json run successfully",
        "production-m/snapshot-corruption/experiment.json run successfully",
        "production-m/stress-cpu-on-broker/experiment.json run successfully",
        "production-m/worker-restart/experiment.json run successfully"
      ]
    }
  }
}


```

pihme commented on core/chaos-workers/handleJobTest.sh:
```diff
@@ -170,3 +184,16 @@ failFunction() {
 }
 
+
+@test "create success message with multiple fail messages" {
```
```suggestion
@test "create success message with multiple result messages" {
```

pihme reviewed (changes requested):
You guessed correctly. There is one tiny change required in Java.

Please add the `SKIPPED` value to `TestReport.TestResult`. Otherwise, the Java code cannot parse the value when trying to write the result to Google Sheet.

Zelldon commented:
It seems that it hasn't worked well with the shell expansion if no file exist. I fixed that with the last commit, now we can correctly skip the execution and mark the job as skipped. 

I tried that out with simple monitor again
```
2020/11/13 09:07:56 Handler completed job 2251799813685653 with variables {
  "testResult": "SKIPPED",
  "testReport": {
    "testResult": "SKIPPED",
    "failureMessages": [],
    "failureCount": 0,
    "metaData": {
      "results": [
        "Skipped test. There were no experiments to run"
      ]
    }
  }
}
```

Furthermore for completeness I added `SKIPPED` to the TestResult enum, but still I think it is not necessary, since the `RecordTestResultHandler` is not used by the chaos experiments as far as I know.

pihme commented:
> RecordTestResultHandler is not used by the chaos experiments as far as I know.

Well, then it should be used. How else do you want to inform the user of the test result?

Zelldon commented:
Slack?

pihme reviewed (approved):
Looks good

pihme commented:
Well, the Slack worker also deserializes the test result.

Btw, is there any particular reason why you do not want to record the test result?

Zelldon commented:
bors r+

Zelldon commented:
Saw no benefit for it, yet. More important was for me to have a notification for slack when it fails first. I mean I only care if it fails, as long as it runs fine I don't care. I don't know whether I want to look at it. Plus I can see all runs also in operate via the variables right?  But If you really want I can create a separate issue for it :)

pihme commented:
Yes please!
